pulp_rpm/app/models.py

```diff
@@ -784,3 +792,270 @@ class ModulemdDefaults(Content):
+        unique_together = (
+            "header_version",
+            "release_name",
+            "release_short",
+            "release_version",
+            "arch",
+            "build_timestamp",
+        )
+
+
+class Checksum(Model):
+    """
+    Distribution Tree Checksum.
+
+    Checksums of selected files in a tree.
+
+    Fields:
+        path (Text):
+            File path
+        checksum (Text):
+            Checksum value for the file
+
+    Relations:
+
+        distribution_tree (models.ForeignKey): The associated DistributionTree
+
+    """
+
+    path = models.CharField(max_length=128)
+    checksum = models.CharField(max_length=128, null=True)
+    distribution_tree = models.ForeignKey(
+        DistributionTree, on_delete=models.CASCADE, related_name='checksums'
+    )
+
+    class Meta:
+        unique_together = (
+            "path",
+            "checksum",
+            "distribution_tree",
+        )
+
+
+class Image(Model):
+    """
+    Distribution Tree Image.
+
+    Images compatible with particular platform.
+
+    Fields:
+        name (Text):
+            File name
+        path (Text):
+            File path
+        platforms (Text):
+            Compatible platforms
+
+    Relations:
+
+        distribution_tree (models.ForeignKey): The associated DistributionTree
+        artifact (models.ForeignKey): The associated Artifact
+
+    """
+
+    name = models.CharField(max_length=20)
+    path = models.CharField(max_length=128)
+    platforms = models.CharField(max_length=20)
+    distribution_tree = models.ForeignKey(
+        DistributionTree, on_delete=models.CASCADE, related_name='images'
+    )
+    artifact = models.ForeignKey(
+        Artifact, on_delete=models.PROTECT, related_name='+'
+    )
+
+    class Meta:
+        unique_together = (
+            "name",
+            "path",
+            "platforms",
+            "distribution_tree",
+        )
+
+
+class Addon(Model):
+    """
+    Distribution Tree Addon.
+
+    Kickstart functionality expansion.
+
+    Fields:
+        addon_id (Text):
+            Addon id
+        uid (Text):
+            Addon uid
+        name (Text):
+            Addon name
+        type (Text):
+            Addon type
+        packages (Text):
+            Relative path to directory with binary RPMs
+
+    Relations:
+
+        distribution_tree (models.ForeignKey): The associated DistributionTree
+        repository (models.ForeignKey): The associated Repository
+
+    """
+
+    addon_id = models.CharField(max_length=50)
+    uid = models.CharField(max_length=50)
+    name = models.CharField(max_length=50)
+    type = models.CharField(max_length=20)
+    packages = models.CharField(max_length=50)
+    distribution_tree = models.ForeignKey(
+        DistributionTree, on_delete=models.CASCADE, related_name='addons'
+    )
+    repository = models.ForeignKey(
+        Repository, on_delete=models.CASCADE, related_name='addons'
```

Comment:
I think this needs to be `on_delete=models.PROTECT` because we can't allow the user to delete a repo that is still backing DistributionTree units in Pulp. Does this make sense?